core/src/main/resources/volcano-viz/viz-template.html

```diff
@@ -0,0 +1,323 @@
+    // Set up an SVG group so that we can translate the final graph.
+    var svg = d3.select("svg");
+    var svgGroup = svg.append("g");
+
+    // Set up zoom support
+    var svg = d3.select("svg")
+        .attr("width", "1200px")
+        .attr("height", "800px")
+        .call(d3.zoom().on("zoom", function () {
+            svgGroup.attr("transform", d3.event.transform)
+        }));
+
+    /*
+     * Global State
+     */
+
+    var currentRuleID = undefined;
+
+    /*
+     * Event Handler functions
+     */
+
+    var setCurrentRule = (ruleMatchID) => {
+        // un-highlight previous entry
+        var prevRuleID = currentRuleID;
+        if (prevRuleID !== undefined) {
+            var prevRuleElement = document.getElementById(prevRuleID);
+            prevRuleElement.style.backgroundColor = "#FFFFFF";
+        }
+
+        currentRuleID = ruleMatchID;
+        document.getElementById('current-rule').innerText = currentRuleID;
+
+        var currentRuleElement = document.getElementById(currentRuleID);
+        currentRuleElement.style.backgroundColor = "#D3D3D3";
+
+        var ruleIndex = ruleMatchSequence.indexOf(currentRuleID);
+
+        document.getElementById("prev-button").disabled = false;
+        document.getElementById("next-button").disabled = false;
+
+        if (ruleIndex === 0) {
+            document.getElementById("prev-button").disabled = true;
+        }
+        if (ruleIndex === ruleMatchSequence.length - 1) {
+            document.getElementById("next-button").disabled = true;
+        }
+
+        createGraph(ruleMatchID);
+    }
+
+    var createGraph = (ruleMatchID) => {
+        var ruleMatchInfo = ruleMatchInfoMap[ruleMatchID]
+        console.log(ruleMatchInfo);
+
+        // remove previous rendered view and clear graph model
+        d3.select("svg g").selectAll("*").remove();
+        g.nodes().slice().forEach(nodeID => g.removeNode(nodeID));
+
+        // create nodes and sets
+        for (var setID in ruleMatchInfo.nodesInSet) {
+            // add set
+            var setLabel = ruleMatchInfo.setLabels[setID];
+            if (setLabel === null || setLabel === undefined) {
+                setLabel = setID;
+            }
+            g.setNode(setID, {
+                label: setLabel,
+                clusterLabelPos: 'top'
+            });
+            // add nodes and node-set parent relationship
+            var nodes = ruleMatchInfo.nodesInSet[setID];
+
+            nodes.forEach(nodeID => {
+                nodeInfo = allNodes[nodeID];
+                var nodeLabel;
+                if (ruleMatchID === "FINAL") {
+                    nodeLabel = nodeInfo.label + "--" + nodeInfo.finalCost;
+                } else {
+                    nodeLabel = nodeInfo.label;
+                }
+                var nodeStyle;
+               if (ruleMatchInfo.importanceZeroNodes
+                    && ruleMatchInfo.importanceZeroNodes.includes(nodeID)) {
+                    nodeStyle = "fill: #D3D3D3"
```

Comment:
Using a class instead of explicit CSS attributes would allow to change the colors later. The same applies to the other styles set by the js code.

Reply:
agree